src/routes/view-content/+page.svelte

```diff
@@ -79,12 +78,14 @@
     let bookChapterSelectorPane: CupertinoPane;
     let cbbterSelectedStepScroll: number | undefined;
     let bibleSelectionScroll: number | undefined;
-    let baseFetchPromise: Promise<void> | undefined;
+    let baseFetchPromise: Promise<[void, void]> | undefined;
     let resourceFetchPromise: Promise<void> | undefined;
     let multiClipAudioStates: Record<string, MultiClipAudioState> = {};
     let preferredBiblesModalOpen = false;
 
-    $: $page.url && fetchBase($selectedBibleSection); // when the Bible section changes, refetch
+    $: {
+        baseFetchPromise = fetchBase($selectedBibleSection); // when the Bible section changes, refetch
```

Comment:
Instead of page.URL && I attached this to the baseFetchPromise, and it is now showing loading screen on bible selection updates.